solnlib/bulletin_rest_client.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright 2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+from solnlib import splunk_rest_client as rest_client
+from splunklib import binding
+from typing import Optional
+import json
+
+
+__all__ = ["BulletinRestClient"]
+
+
+class BulletinRestClient:
+    READ_ENDPOINT = "/services/admin/messages"
+    CREATE_DELETE_ENDPOINT = "/services/messages"
+
+    headers = [("Content-Type", "application/json")]
+
+    class LogLevel:
+        INFO = "info"
+        WARNING = "warn"
+        ERROR = "error"
+
+    def __init__(
+        self,
+        message_name: str,
+        session_key: str,
+        logger: Optional[logging.Logger] = None,
+        **context: dict,
+    ):
+
+        self.message_name = message_name
```

Comment:
There will be one value per one instance. So we will need to create a second instance to add a new message?

Reply:
Yes. These messages work in this way: when creating a message, we must provide a name, which is some kind of ID. When we try to create another message with the same name (ID), the system will not add another message to the bulletin, but will overwrite the existing one. Same for deletion. To delete a message, we must indicate the name (ID). Therefore, we have two options: the first one I chose is to create instances per message. I think this will make it easier for the user to control a specific message. The second option is to make the client more generic and then, each time the user uses create, get, delete, the user will also have to provide the message (ID) for which he wants to execute a given command. It seems to me that the second option is more susceptible to human errors, etc., although it can be easily changed, so if you think that we should go with the second option here, let me know in the comment

Reply:
Can you please put this explanation into the docstring? I think developer will benefit of your explanation a lot!

Reply:
added